src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSslVersion.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class OpenSsl
+    {
+        private static Version s_opensslVersion;
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_SSLEayVersion")]
+        private static extern string OpenSslVersionDescription();
+
+        internal static Version OpenSslVersion
+        {
+            get
+            {
+                if (s_opensslVersion == null)
+                {
+                    const string OpenSSL = "OpenSSL ";
+
+                    // Skip OpenSSL part, and get the version string of format x.y.z
+                    if (!Version.TryParse(OpenSslVersionDescription().Substring(OpenSSL.Length, 5), out s_opensslVersion))
+                    {
+                        s_opensslVersion = new Version(0, 0, 0);
```

Comment:
Have you confirmed that the tests are running everywhere we want them to?

Reply:
I was able to confirm locally on my ubuntu 17.04 box, where the openssl version is 1.0.2 and the test parses the version correctly and executes the alpn tests.